test/common/quic/envoy_quic_h3_fuzz_test.cc

```diff
@@ -0,0 +1,768 @@
+public:
+  FuzzDispatcher(quic::ParsedQuicVersion quic_version, quic::QuicVersionManager* version_manager,
+                 std::unique_ptr<quic::QuicConnectionHelperInterface> connection_helper,
+                 std::unique_ptr<quic::QuicAlarmFactory> alarm_factory,
+                 Event::Dispatcher& dispatcher)
+      : quic::QuicDispatcher(
+            &quic_config_, &crypto_config_, version_manager, std::move(connection_helper),
+            std::make_unique<EnvoyQuicCryptoServerStreamHelper>(), std::move(alarm_factory),
+            quic::kQuicDefaultConnectionIdLength, generator_),
+        listener_config_(&mock_listener_config_),
+        quic_stats_(generateStats(listener_config_->listenerScope())), dispatcher_(dispatcher),
+        http3_options_(http3Settings()), quic_version_(quic_version),
+        packetizer_(quic_version_, helper()),
+        crypto_config_(quic::QuicCryptoServerConfig::TESTING, quic::QuicRandom::GetInstance(),
+                       std::make_unique<TestProofSource>(), quic::KeyExchangeSource::Default()),
+        peer_addr_(Network::Utility::getAddressWithPort(*Network::Utility::getIpv6LoopbackAddress(),
+                                                        12345)),
+        self_addr_(Network::Utility::getAddressWithPort(*Network::Utility::getIpv6LoopbackAddress(),
+                                                        54321)),
+        cli_addr_(peer_addr_->sockAddr(), peer_addr_->sockAddrLen()),
+        srv_addr_(self_addr_->sockAddr(), self_addr_->sockAddrLen()),
+        quic_stat_names_(listener_config_->listenerScope().symbolTable()),
+        http3_stats_({ALL_HTTP3_CODEC_STATS(
+            POOL_COUNTER_PREFIX(listener_config_->listenerScope(), "http3."),
+            POOL_GAUGE_PREFIX(listener_config_->listenerScope(), "http3."))}),
+        connection_stats_({QUIC_CONNECTION_STATS(
+            POOL_COUNTER_PREFIX(listener_config_->listenerScope(), "quic.connection"))}) {
+    ON_CALL(http_connection_callbacks_, newStream(_, _))
+        .WillByDefault(Invoke([&](Http::ResponseEncoder&, bool) -> Http::RequestDecoder& {
+          return orphan_request_decoder_;
+        }));
+    auto writer = new testing::NiceMock<quic::test::MockPacketWriter>();
+    ON_CALL(*writer, WritePacket(_, _, _, _, _))
+        .WillByDefault(testing::Return(quic::WriteResult(quic::WRITE_STATUS_OK, 0)));
+    InitializeWithWriter(writer);
+  }
+
+  void fuzzQuic(const QuicH3FuzzCase& input) {
+    quic::QuicReceivedPacket first_packet(nullptr, 0, helper()->GetClock()->Now());
+    quic::ParsedClientHello chlo;
+    quic::ReceivedPacketInfo chlo_packet_info(srv_addr_, cli_addr_, first_packet);
+    chlo_packet_info.destination_connection_id = quic::test::TestConnectionId();
+    chlo_packet_info.version = quic_version_;
+    SetQuicFlag(quic_allow_chlo_buffering, false);
+    ProcessChlo(chlo, &chlo_packet_info);
+
+    packetizer_.serializePackets(input);
+    for (size_t i = 0; i < kMaxNumPackets; i++) {
```

Comment:
Is there a reason why the number of iterations is up to the const and not relative to the number of entries in the input?

Reply:
no reason, changed to the expectable number of inputs.